src/words/environment/WordsEnvironment.java

```diff
@@ -15,17 +16,18 @@
 	/* gettableObjects is a list of locally scoped symbol tables with the most local 
 	 * scope first in the list, and global scope always being last in the list.
 	 */
-	private LinkedList<HashMap<String, WordsObject>> gettableObjects;
-	/* ungettableObjects contains the objects that have gone out of scope, but should still appear
-	 * on the board, and are still used by action listeners */
-	private HashSet<WordsObject> ungettableObjects;
+	private LinkedList<HashMap<String, WordsObject>> objectsByName;
```

Comment:
I like this refactor a lot.  I think objectsByName is much clearer than gettableObjects.